### DISCUSSION TITLE
Have you seen delays to ACK TCP packets?

### DISCUSSION THREAD

**Discussion by DavidVentura** [CarlosDerSeher/snapclient#71] [Q&A]
Hi!
I've been building a snapclient for the esp32 as well ([here](https://github.com/DavidVentura/esp-snapcast)) and I've been getting "dropouts" seemingly every 70 minutes.
The way that I observe these "dropouts" is that the ESP32 will not ACK a packet, causing a retransmission on the snapserver, which delays the next WireChunk by 600ms; sadly, 600ms is larger than my current buffer :cry: and this causes a gap in playback

My question is, have you experienced similar behavior? I've been poking around your `sdkconfig` file and checking for any possible tunable; I've moved WIFI and IP code to IRAM, increased the TCP and WIFI RX mboxes, etc.

**Comment by CarlosDerSeher** [comment]
No dropouts here. 600ms seems like a very long time to me and I don't think this is a TCP problem. Did you rule out your access point / internet / server as the source of error?

**Reply by DavidVentura** [reply]
Well, I don't know if it's my access point.
I have a packet capture which shows that the ACK was received 571ms after the initial packet:
![image](https://github.com/CarlosDerSeher/snapclient/assets/3650670/3170de1e-ab4f-4288-b30a-993b02f81721)
![image](https://github.com/CarlosDerSeher/snapclient/assets/3650670/a4b74381-ca84-4615-9eb6-95039b698495)

I'm fishing a bit at this point, as I'm out of leads with my implementation.

Have you tested your implementation of `snapclient` with a 500ms (or smaller) buffer for extended periods of time? (>1h)

**Reply by CarlosDerSeher** [reply]
I actually set it to 500ms recently but I am not at home much so I didn't use snapclient very often. I see if I can try this weekend. Normally I use it with >700ms. What you can try, is to print rssi and your buffer status to the log so you can see when it is happening.

Do you have any other traffic on your esp32? From your Readme I am guessing you are using ntp for synchronization? Maybe your problem lies there? Try to disable it to see if things work out then...

**Reply by CarlosDerSeher** [reply]
@DavidVentura did you resolve your issue? Are you using sntp for syncing in some way? I don't have any dropouts in my network even if just using 500ms buffers.
